sky/policy.py

```diff
@@ -1,154 +1,82 @@
-"""Customize policy by users."""
-import copy
+"""Interface for admin-defined policy for user requests."""
+import abc
 import dataclasses
-import importlib
-import os
-import tempfile
 import typing
-from typing import Any, Callable, Dict, Optional
+from typing import Optional
 
-from sky import dag as dag_lib
-from sky import sky_logging
-from sky import skypilot_config
-from sky.utils import common_utils
-from sky.utils import ux_utils
+if typing.TYPE_CHECKING:
+    import sky
 
-logger = sky_logging.init_logger(__name__)
 
-if typing.TYPE_CHECKING:
-    from sky import task as task_lib
+@dataclasses.dataclass
+class OperationArgs:
+    cluster_name: Optional[str]
+    cluster_exists: bool
+    idle_minutes_to_autostop: Optional[int]
+    down: bool
+    dryrun: bool
 
 
 @dataclasses.dataclass
-class UserTask:
-    task: 'task_lib.Task'
-    skypilot_config: Dict[str, Any]
+class UserRequest:
+    """User request to the policy.
+
+    Args:
+        task: User specified task.
+        skypilot_config: Global skypilot config.
+        execution_args: Execution arguments. It can be None for jobs and
+            services.
+    """
+    task: 'sky.Task'
```

Comment:
Actually, does this cover `sky jobs launch` (a pipeline; which is more than a singleton task) and `sky serve up`?

Reply:
Yes, it covers `jobs launch` and `sky serve up` (see the test in PR description). We apply the policy locally to the task and config, and upload it to the remote controller for execution.